### PR TITLE
Create EFS storage guide for AWS

### PR DESCRIPTION
**Pull request by cam72cam**
For https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/421

### CODE REVIEW

**Comment by summerswallow-whi** [comment]
Have you done anything with encryption in transit? (e.g. NFS over TLS)

**Comment by cam72cam** [comment]
I have not, not sure how that works with EFS

**Comment by cam72cam** [comment]
https://docs.aws.amazon.com/efs/latest/ug/encryption.html#encryption-in-transit might be a good read, but integrating that into K8S might be a challenge

File: doc/source/amazon/efs_storage.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+   Run these commands to setup your namespace and storage::
+   
+       kubectl create namespace <your namespace>
+       kubectl --namespace=<your namespace> apply -f test_efs.yaml
+       kubectl --namespace=<your namespace> apply -f test_efs_claim.yaml
+
+
+   I don't know if the PV needs to be in the namespace, but the arg does not seem to hurt anything.  The PVC must be in the namespace or stuff will break in weird ways.
+
+3. Configuring your application to use EFS as it's backing storage
```
Comment (willingc) [inline doc/source/amazon/efs_storage.rst]: its

File: doc/source/amazon/efs_storage.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+   We now add the following to config.yaml::
```
Comment (willingc) [inline doc/source/amazon/efs_storage.rst]: Let's put config.yaml in backticks.

```
`config.yaml`
```

File: doc/source/amazon/efs_storage.rst [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+             subPath: 'home/{username}'
+         extraEnv:
+           CHOWN_HOME: 'yes'
+         uid: 0
+         fsGid: 0
+         cmd: "start-singleuser.sh"
+
+
+   The image setting overrides the default pinned jh base image since it has not yet been updated 
+   to include the CHOWN_HOME setting.  This will be fixed in Z2JH 0.7.
```
Comment (willingc) [inline doc/source/amazon/efs_storage.rst]: Zero to JupyterHub v0.7

Let's expand jh to JupyterHub in several places

File: doc/source/amazon/efs_storage.rst [hunk 4]
```diff
@@ -0,0 +1,128 @@
+.. _amazon-aws:
+
+Setting up EFS storage on AWS
+-----------------------------
+
```
Comment (willingc) [inline doc/source/amazon/efs_storage.rst]: Add:

```
**Note: This section is experimental. Do not use in production unless you fully understand the drawbacks and limitations.**

**Comment by willingc** [comment]
https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/707#issuecomment-396407004

Thanks @cam72cam.

**Pull request merged**